src/Infrastructure.EntityFramework/AdminConsole/Repositories/OrganizationRepository.cs

```diff
@@ -270,8 +271,199 @@ into grouped
+                    })
+                    .ToList();
+                dbContext.CollectionUsers.AddRange(newCollectionUsers);
+            }
+
+            // Update OrganizationUser to clear AccessAll flag
+            dbContext.OrganizationUsers
+                .Where(ou => orgUserIdsWithAccessAll.Contains(ou.Id))
+                .ToList()
+                .ForEach(ou => ou.AccessAll = false);
+
+            // Save changes for Step 2
+            await dbContext.SaveChangesAsync();
+
+            // Step 3: For all OrganizationUsers with Manager role or 'EditAssignedCollections' permission
+            // update their existing CollectionUser rows and insert new rows with [Manage] = 1
+            // and finally update all OrganizationUsers with Manager role to User role
+            var managerOrgUsersIds = dbContext.OrganizationUsers
+                .Where(ou =>
+                    ou.OrganizationId == organizationId &&
+                    (ou.Type == OrganizationUserType.Manager ||
+                     (ou.Permissions != null && EF.Functions.Like(ou.Permissions, "%\"editAssignedCollections\":true%"))))
+                .Select(ou => ou.Id)
+                .ToList();
+
+            // Update CollectionUser rows with Manage = true
+            dbContext.CollectionUsers
+                .Where(cu => managerOrgUsersIds.Contains(cu.OrganizationUserId))
+                .ToList()
+                .ForEach(cu =>
+                {
+                    cu.ReadOnly = false;
+                    cu.HidePasswords = false;
+                    cu.Manage = true;
+                });
+
+            // Insert rows into CollectionUser with Manage = true
+            foreach (var manager in managerOrgUsersIds)
+            {
+                var newCollectionUsersWithManage = (from cg in dbContext.CollectionGroups
+                                                    join gu in dbContext.GroupUsers on cg.GroupId equals gu.GroupId
+                                                    where gu.OrganizationUserId == manager &&
+                                                          !dbContext.CollectionUsers.Any(cu =>
+                                                              cu.CollectionId == cg.CollectionId &&
+                                                              cu.OrganizationUserId == manager)
+                                                    select new CollectionUser
+                                                    {
+                                                        CollectionId = cg.CollectionId,
+                                                        OrganizationUserId = manager,
+                                                        ReadOnly = false,
+                                                        HidePasswords = false,
+                                                        Manage = true
+                                                    }).Distinct().ToList();
```

Comment:
I wasn't sure about mixing the linq query and method syntax, but this is a bit more complex so I can understand why you've done it. Some things are easier to express in the query syntax.

Reply:
I don't like mixing them either but I'm always for readability first. I think method syntax works best for simple queries whereas query syntax is better for more complex queries (maybe because its similar to SQL).